packages/flutter/lib/src/material/text_selection.dart

```diff
@@ -20,64 +23,411 @@ const double _kHandleSize = 22.0;
+      }
+    });
+
+    // If the last child overflows, but only because of the width of the
+    // overflow button, then just show it and hide the overflow button.
+    final RenderBox navButton = firstChild;
+    if (_lastIndexThatFits != -1 && _lastIndexThatFits == childCount - 2
+      && width - navButton.size.width <= constraints.maxWidth) {
+      _lastIndexThatFits = -1;
+    }
+  }
+
+  // Set the offset of all of the children that will be painted.
+  void _placeChildren() {
+    int i = -1;
+    Size nextSize = const Size(0.0, 0.0);
+    double fitWidth = 0.0;
+    final RenderBox navButton = firstChild;
+    double overflowHeight = overflowOpen && !isAbove ? navButton.size.height : 0.0;
+    visitChildren((RenderObject renderObjectChild) {
+      i++;
+
+      // The navigation button is placed after iterating all children.
+      if (renderObjectChild == firstChild) {
+        return;
+      }
+
+      // No need to place children that won't be painted.
+      if (!_shouldPaintChild(renderObjectChild, i)) {
+        return;
+      }
+
+      final RenderBox child = renderObjectChild as RenderBox;
+      final FlexParentData childParentData = child.parentData as FlexParentData;
+
+      if (!overflowOpen) {
+        childParentData.offset = Offset(fitWidth, 0.0);
+        fitWidth += child.size.width;
+        nextSize = Size(
+          fitWidth,
+          math.max(child.size.height, nextSize.height),
+        );
+      } else {
+        childParentData.offset = Offset(0.0, overflowHeight);
+        overflowHeight += child.size.height;
+        nextSize = Size(
+          math.max(child.size.width, nextSize.width),
+          overflowHeight,
+        );
+      }
+    });
+
+    // Place the navigation button if there is overflow.
+    if (_lastIndexThatFits >= 0) {
+      final FlexParentData navButtonParentData = navButton.parentData as FlexParentData;
+      if (overflowOpen) {
+        navButtonParentData.offset = isAbove
+          ? Offset(0.0, overflowHeight)
+          : Offset.zero;
+        nextSize = Size(
+          nextSize.width,
+          isAbove ? nextSize.height + navButton.size.height : nextSize.height,
+        );
+      } else {
+        navButtonParentData.offset = Offset(fitWidth, 0.0);
+        nextSize = Size(nextSize.width + navButton.size.width, nextSize.height);
+      }
+    }
+
+    size = nextSize;
+  }
+
+  // Returns true when the child should be painted, false otherwise.
+  bool _shouldPaintChild(RenderObject renderObjectChild, int index) {
+    // Paint the navButton when there is overflow.
+    if (renderObjectChild == firstChild) {
+      return _lastIndexThatFits != -1;
+    }
+
+    // If there is no overflow, all children besides the navButton are painted.
+    if (_lastIndexThatFits == -1) {
+      return true;
+    }
+
+    // When there is overflow, paint if the child is in the part of the menu
+    // that is currently open.
+    return (index > _lastIndexThatFits) == overflowOpen;
```

Comment:
when the overflow is open and index<= _lastIndexThatFits, don't we still want to paint?

Reply:
No, because when the overflow menu is open, only the overflowing items are shown.  So only the items with index > _lastIndexThatFits.  The opposite items are shown when the overflow menu is closed.

I'll add some more detail in my comments for overflowOpen because this is pretty nuanced.